graphql_client_codegen/src/lib.rs

```diff
@@ -44,67 +43,92 @@ impl std::error::Error for GeneralError {}
+        }
+    })
+}
+
+/// Generates Rust code given a path to a query file, a path to a schema file, and options.
 pub fn generate_module_token_stream(
     query_path: std::path::PathBuf,
     schema_path: &std::path::Path,
     options: GraphQLClientCodegenOptions,
 ) -> Result<TokenStream, BoxError> {
-    use std::collections::btree_map;
-
-    let schema_extension = schema_path
-        .extension()
-        .and_then(std::ffi::OsStr::to_str)
-        .unwrap_or("INVALID");
-    let schema_string;
-
-    // Check the schema cache.
-    let schema: schema::Schema = {
-        let mut lock = SCHEMA_CACHE.lock().expect("schema cache is poisoned");
-        match lock.entry(schema_path.to_path_buf()) {
-            btree_map::Entry::Occupied(o) => o.get().clone(),
-            btree_map::Entry::Vacant(v) => {
-                schema_string = read_file(v.key())?;
-                let schema = match schema_extension {
-                    "graphql" | "gql" => {
-                        let s = graphql_parser::schema::parse_schema::<&str>(&schema_string).map_err(|parser_error| GeneralError(format!("Parser error: {}", parser_error)))?;
-                        schema::Schema::from(s)
-                    }
-                    "json" => {
-                        let parsed: graphql_introspection_query::introspection_response::IntrospectionResponse = serde_json::from_str(&schema_string)?;
-                        schema::Schema::from(parsed)
-                    }
-                    extension => return Err(GeneralError(format!("Unsupported extension for the GraphQL schema: {} (only .json and .graphql are supported)", extension)).into())
-                };
-
-                v.insert(schema).clone()
-            }
-        }
-    };
+    let query = get_set_query_from_file(query_path.as_path());
+    let schema = get_set_schema_from_file(schema_path);
 
-    // We need to qualify the query with the path to the crate it is part of
-    let (query_string, query) = {
-        let mut lock = QUERY_CACHE.lock().expect("query cache is poisoned");
-        match lock.entry(query_path) {
-            btree_map::Entry::Occupied(o) => o.get().clone(),
-            btree_map::Entry::Vacant(v) => {
-                let query_string = read_file(v.key())?;
-                let query = graphql_parser::parse_query(&query_string)
-                    .map_err(|err| GeneralError(format!("Query parser error: {}", err)))?
-                    .into_static();
-                v.insert((query_string, query)).clone()
-            }
-        }
-    };
+    generate_module_token_stream_inner(&query, &schema, &options)
+}
 
-    let query = crate::query::resolve(&schema, &query)?;
+/// Generates Rust code given a query string, a path to a schema file, and options.
+pub fn generate_module_token_stream_from_string(
+    query_string: &str,
+    schema_path: &std::path::Path,
+    options: &GraphQLClientCodegenOptions,
```

Comment:
I suppose I should keep this consistent?

```suggestion
    options: GraphQLClientCodegenOptions,
```